src/java.base/windows/classes/sun/nio/fs/WindowsPathParser.java

```diff
@@ -147,6 +161,14 @@ private static Result parse(String input, boolean requireToNormalize) {
             }
         }
 
+        if (type != expectedType) {
```

Comment:
Maybe this should also have `expectedType != null` either before of after the existing check?

Reply:
Yes, `if (expectedType != null && type != expectedType)` would be clearer.

Reply:
So changed in Path-long-prefix-8298478.